errors.go

```diff
@@ -308,6 +308,8 @@ func Propagate(err error) error {
 // Is checks whether an error is a given code. Similarly to `errors.Is`,
 // this unwinds the error stack and checks each underlying error for the code.
 // If any match, this returns true.
+// Note that Is only behaves differently to PrefixMatches when errors in the stack have different codes.
+// This is the case when errors are initialized with NewInternalWithCause, but not with Augment.
 // We prefer this over using a method receiver on the terrors Error, as the function
 // signature requires an error to test against, and checking against terrors would
 // requite creating a new terror with the specific code.
```

Comment:
I think it would be good to remove the "experimental" warnings on these functions now (I think there's one on `Propagate` and maybe `Augment` too). Happy for that to be a separate PR, or this one, but we're pretty set on these now.

Reply:
I can do this in a separate PR.

Reply:
I was just trying to push my branch `remove-experimental-warnings` - looks like this already exists! https://github.com/monzo/terrors/pull/31